charms/jupyter-ui/src/charm.py

```diff
@@ -184,13 +263,24 @@ def _get_interfaces(self):
             raise CheckFailed(str(err), BlockedStatus)
         return interfaces
 
-    def _check_image_details(self):
+    def main(self, _) -> None:
```

Comment:
This might be a bit nitpicky, but given that we now have different event handlers besides `main` I would give this method a different name. Something like `reconfigure` for example, although I'm definitely not married to that name.